src/escalus_tcp.erl

```diff
@@ -85,7 +86,8 @@
 -spec connect([proplists:property()] | opts()) -> pid().
 connect(Opts0) ->
     Opts1 = opts_to_map(Opts0),
-    {ok, Pid} = gen_server:start_link(?MODULE, [Opts1, self()], []),
+    GenOpts = maps:to_list(maps:with([hibernate_after], Opts1)),
```

Comment:
shouldn't we do `overwrite_default_opts/2`call here instaed of `init/1`, so default value for `hibernate_after` is applied to `gen_server`